packages/blockchain-utils-linking/src/ethereum.ts

```diff
@@ -61,10 +63,40 @@ export class EthereumAuthProvider implements AuthProvider {
     return createLink(did, accountId, this.provider, this.opts)
   }
 
-  async requestCapability(siweMessage: SiweMessage): Promise<Cacao> {
+  async requestCapability(
+    sessionDID: string,
+    streams: Array<StreamID | string>,
+    opts: CapabilityOpts
+  ): Promise<Cacao> {
     console.warn(
       'WARN: requestCapability os an experimental API, that is subject to change any time.'
     )
+
+    const domain = typeof window !== 'undefined' ? window.location.hostname : opts.domain
+    if (!domain) throw new Error("Missing parameter 'domain'")
+
+    // NOTE: To allow proper customization of the expiry date, we need a solid library to represent
+    // time durations that includes edge cases. We should not try dealing with timestamps ourselves.
+    const now = new Date()
+    const oneWeekLater = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000)
+
+    const siweMessage = new SiweMessage({
+      domain: domain,
+      address: this.address,
+      statement: opts.statement
+        ? opts.statement
+        : 'Give this application access to some of your data on Ceramic',
+      uri: sessionDID,
+      version: opts.version ? opts.version : '1',
+      nonce: opts.nonce ? opts.nonce : randomString(10),
+      issuedAt: now.toISOString(),
+      expirationTime: oneWeekLater.toISOString(),
+      chainId: (await this.accountId()).chainId.toString(),
+      resources: streams.map((s) => (typeof s === 'string' ? s : s.toUrl())),
```

Comment:
```suggestion
      expirationTime: opts.expirationTime || oneWeekLater.toISOString(),
      chainId: (await this.accountId()).chainId.toString(),
      resources: (opts.resources || []).concat(streams.map((s) => (typeof s === 'string' ? s : s.toUrl()))),
```

Reply:
I did not add `expirationTime` to opts because what if someone creates invalid expiry dates? Also, they are now responsible for formatting the string the way we want. Invalid formatting might still work but lead to weird results.

I suggested using a proper datetime library to handle time durations properly instead of exact timestamps of expiry.

Reply:
I expect people will request this feature if we don't add it now. But we don't need to include it in this PR.